hpc0.qmd

```diff
@@ -7,279 +7,1223 @@ format:
+- octal value = read (4) + write (2) + execute (1)
+- A later exercise will require adding execute permissions for a file...
 :::
 
-## Exercise 6 {.smaller}
+
+## Groups
+- System users belong to groups (at least one)
+- A user can change the group assignment for a file or directory to other groups they belong to using the command `chgrp`
+- The command `groups` can be used to print out what groups there are, and what group a user is part of
+- This can be useful if you want to share files between those in a group on the system
+
+
+## Check point 3
+- Any questons?
+
+::: {.notes}
+- Time for a break?
+:::
+
+
+## Variables
+- Variables are defined and accessed as follows:
+```{.bash code-line-numbers=false}
+var=1
+echo $var
+```
+- The variable `var` is set equal to the number `1`.
+- `echo` is a command that will send what comes after it to the Standard Output (stdout).
+- The value of the variable is accessed using the `$` symbol followed by the name of the variable.
+- The command `printenv` will print all the variables in use.
+
+
+## For loop 1/2
+### Examples
+- Iterator syntax to print the numbers 1 to 5:
+```{.bash code-line-numbers=false}
+for ((i=1;i<=5;i=i+1)) do echo $i; done
+```
+- Sequence syntax to print the numbers 1 to 5:
+```{.bash code-line-numbers=false}
+for i in {1..5}; do echo $i; done
+```
+- Sequence syntax to print even numbers from 2 to 10:
+```{.bash code-line-numbers=false}
+for i in {2..10..2}; do echo $i; done
+```
+- Iterator syntax to print a geometric series from 2 to 256:
+```{.bash code-line-numbers=false}
+for ((i=2;i<=265;i=i*2)) do echo $i; done
+```
+
+
+## For loop 2/2
+- Example for loop to loop through directory contents:
+```{.bash code-line-numbers=false}
+for f in *; do echo $f; done
+```
+- Where there is a `;` there can be a new line:
+```{.bash code-line-numbers=false}
+for f in *
+do echo $f
+done
+```
+
+
+## For loops exercise 1
 
 ::: {.panel-tabset}
 
 ### Problem
-
-1. In the wildcards directory, create a variable called files listing all of the text files. 
-2. Loop through this list and print out the first line from each file.
+- Use a for loop to create five directories called `calculation_?`, where `?` is a number.
 
 ### Solution
+- For loop with an iterator syntax
+```{.bash code-line-numbers=false}
+for ((i=1;i<6;i++))
+do
+mkdir calculation_$i
+done
+```
+- `i++` is shorthand for `i=i+1`
+- For loop with a sequence syntax
+```{.bash code-line-numbers=false}
+for i in {1..5}
+do
+mkdir calculation_$i
+done
+```
 
-1. ```{.bash code-line-numbers=false}
-[user@host wildcards]$ files=$( ls *.txt )
+:::
+
+
+## For loops exercise 2
+
+::: {.panel-tabset}
+
+### Problem
+- Use a for loop to create five directories, each one the parent of the next.
+
+### Solution
+```{.bash code-line-numbers=false}
+for i in {1..5}; do
+mkdir calculation_$i
+cd calculation_$i
+done
 ```
 
-2. ```{.bash code-line-numbers=false}
-[user@host wildcards]$ for f in $files
-> do
-> head -1 $f
-> done
+:::
```

Comment:
This slide doesn't all fit on the screen

Reply:
I can't see to see the problem...